src/utils/__tests__/use-do-once-when.test.ts

```diff
@@ -78,11 +78,55 @@ describe('useDoOnceWhen', () => {
     expect(functionRunCount).toBe(0);
 
     hook.rerender({condition: true});
-    expect(functionRunCount).toBe(1); // Function runs once
+    expect(functionRunCount).toBe(1);
 
     hook.rerender({condition: false});
 
     hook.rerender({condition: true});
-    expect(functionRunCount).toBe(2); // Function runs again
+    expect(functionRunCount).toBe(2);
+  });
+
+  it('Should rerun when onlyOnce changes from true to false', async () => {
+    const hook = renderHook(
+      ({condition, onlyOnce}) =>
+        useDoOnceWhen(() => functionRunCount++, condition, onlyOnce),
+      {initialProps: {condition: false, onlyOnce: true}},
+    );
+
+    expect(functionRunCount).toBe(0);
+
+    hook.rerender({condition: true, onlyOnce: true});
+    expect(functionRunCount).toBe(1);
+
+    hook.rerender({condition: false, onlyOnce: false});
+    expect(functionRunCount).toBe(1);
+
+    hook.rerender({condition: true, onlyOnce: false});
+    expect(functionRunCount).toBe(2);
```

Comment:
What about:
```ts
    hook.rerender({condition: true, onlyOnce: true});
    expect(functionRunCount).toBe(1);

    hook.rerender({condition: true, onlyOnce: false});
    expect(functionRunCount).toBe(1);

    hook.rerender({condition: true, onlyOnce: true});
    expect(functionRunCount).toBe(?);
```

Reply:
The `?` should be `2`, because the "do only once" starts counting from the render where `onlyOnce` is `true`. Extra test added.

Reply:
Note that the count could be very high after `onlyOnce` has been `false`, it is not possible or meaningful to "go back" to insist it should have only ran once previously..

Reply:
Does it really make sense? Isn't kinda the core of the hook that it only run once when the `condition` stays `true`? 🤔 
- If `onlyOnce` is `false`, then the function is only run once as long as the `condition` stays `true`.
- If `onlyOnce` is `true`, then the function is only run once as long as the `condition` stays `true`.
- But in the case that `onlyOnce` switches between `false` and `true`, then the function is run multiple times even if `condition` always stays `true`.

I'm kinda nitpicking on this one, the actual case here isn't really important. But the subject touches upon something that we have some problems with in the app, that several effects run to often and in unwanted/unnecessary situations.

Reply:
Since switching `onlyOnce` from `false` to `true` is an impossible ask if it has already been run twice or more, I think [this approach](https://github.com/AtB-AS/mittatb-app/pull/4745/files#diff-eebed2193d07efdf49b24bf8d3a425c552998752642ff0cbccaacdb89aa0d491) is preferable.

Reply:
I would rather do the same with `onlyOnce` as with the given function, transform it to a ref so it doesn't affect the `useEffect`.

Reply:
That works, even though the issues noted before here remain 😅 
New PR created: https://github.com/AtB-AS/mittatb-app/pull/4746